wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsDataSource.swift

```diff
@@ -73,7 +73,8 @@ final class ContactsDataSource: NSObject {
     private func performSearch() {
         guard let searchDirectory else { return }
 
-        let request = SearchRequest(query: searchQuery, searchOptions: [.contacts, .addressBook])
+        let selfUserDomain = SelfUser.provider?.providedSelfUser.domain
+        let request = SearchRequest(query: searchQuery, searchDomain: selfUserDomain, searchOptions: [.contacts, .addressBook])
```

Comment:
Since we pass in the self user domain as the search domain, does that mean the search will not return federated results? Even if federation is allow?

Reply:
I've added federation search check to the `ContactsViewController`, but we have a ticket https://wearezeta.atlassian.net/browse/WPB-14654 to hide "Invite more people" button that navigates to this screen